pkg/install/install.go

```diff
@@ -14,12 +15,24 @@ import (
 	"github.com/spf13/cobra"
 )
 
+// PreRunEInstall generate an error if a user wrongly inputs options with –
+// See https://github.com/noobaa/noobaa-operator/issues/484
+func PreRunEInstall(cmd *cobra.Command, args []string) error {
+	flagArgs := cmd.Flags().Args()
+	if len(flagArgs) > 0 {
```

Comment:
I'm less familiar with cobra, but isn't there a more generic solution for that problem? this is probably an issue that happens in other commands as well. for example this will not handle `noobaa install yaml –pv-pool-default-storage-class='<storage class>'`

Reply:
@dannyzaken initially tried to define the PreRunE() filter on the root command, however it breaks commands which do expect non-flag arguments, like bucket name in  `noobaa bucket create <bucket-name> [flags] [options]`

it is also possible to define the PreRunE() fitter on the root command and then exclude the commands which expect the arguments.

WDYT?

Reply:
@dannyzaken , Let's another way. Defined that all noobaa commands should generate an error on non-flag arguments at the root Cmd, children of this command will inherit and execute [PersistentPreRunE](https://github.com/spf13/cobra/blob/4590150168e93f4b017c6e33469e26590ba839df/command.go#L107-L108):

https://github.com/baum/noobaa-operator/blob/bb9102de598eb83f45ca0eb91f58665c2b53ea83/pkg/cli/cli.go#L85-L93

Defined a shared Pass function at options, i.e. this implementation never returns an error:

https://github.com/baum/noobaa-operator/blob/bb9102de598eb83f45ca0eb91f58665c2b53ea83/pkg/options/options.go#L116-L122

Found 23 commands which require non-flag arguments, those commands set their PersistentPreRunE to the passing function, see for instance 'bucket create':
https://github.com/baum/noobaa-operator/blob/bb9102de598eb83f45ca0eb91f58665c2b53ea83/pkg/bucket/bucket.go#L36

So now:
```
➜  noobaa install yaml  –pv-pool-default-storage-class='<storage class>'
Error: unknown args: [–pv-pool-default-storage-class=<storage class>]
```

On the other hand commands requiring non-flag args are happy:
```
➜  noobaa bucket create                  
FATA[0000] Missing expected arguments: <bucket-name> 

Usage:
  noobaa bucket create <bucket-name> [flags] [options]

Use "noobaa options" for a list of global command-line options (applies to all commands). 
➜  noobaa bucket create second.bucket
INFO[0000] ✅ Exists: NooBaa "noobaa"                    
INFO[0000] ✅ Exists: Service "noobaa-mgmt"              
INFO[0000] ✅ Exists: Secret "noobaa-operator"           
INFO[0000] ✅ Exists: Secret "noobaa-admin"              
....
```

How does it look to you?

Reply:
I think it's a good solution, but it means that anyone that is implementing a new command requiring non-flag args will need to be aware of this change, right?
we also need to make sure we didn't miss any of the commands that require non-flag args

Reply:
@dannyzaken, you're correct, commands requiring non-flag arguments need to define the PersistentPreRunE method in such a model, however, the implementation is shared. it was relatively easy to locate those commands requiring non-flag arguments since they reference the args list at the beginning of their Run method.  

cobra commands form a tree, and "persistent" pre-post hooks are inherited. it would be nice if we could mark whole subtrees - that would make our lives easier. however, some tree leaves do not match their sub-tree behavior.  for instance, in the "noobaa bucket" sub-tree there are four subcommands: 
-  create      Create a NooBaa bucket
-  delete      Delete a NooBaa bucket
-  list        List NooBaa buckets
-  status      Show the status of a NooBaa bucket 

and most require a non-flag argument, however `list` does not.

so I struggle to find a better strategy other than defining a default behavior, ie. allow or deny and then mark those commands that do not match.  i do not have a preference either way.

WDYT?